src/cdk/platform/features/shadow-dom.ts

```diff
@@ -0,0 +1,19 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+let shadowDomIsSupported: boolean;
+
+/** Checks whether the user's browser support Shadow DOM. */
+export function supportsShadowDom(): boolean {
```

Comment:
I'm a little hesitant to add this as a public API since we'll probably drop support for IE11 in early 2020, at which point I don't think it would be necessary any more.

Reply:
I wasn't sure where to put it. It's a bit too generic to have inside a component.

Reply:
Let's just give it an underscore prefix